x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -312,50 +330,54 @@ export class ReportingStore {
         if_seq_no: report._seq_no,
         if_primary_term: report._primary_term,
         refresh: true,
-        body: { doc: { process_expiration: null } },
+        body: { doc },
       });
-
-      return (body as unknown) as ReportDocument;
+      return body;
     } catch (err) {
-      this.logger.error('Error in clearing expiration!');
+      this.logger.error(
+        `Error in clearing expiration and status for retry! Report: ` + jobDebugMessage(report)
+      );
       this.logger.error(err);
       throw err;
     }
   }
 
   /*
-   * A zombie report document is one that isn't completed or failed, isn't
-   * being executed, and isn't scheduled to run. They arise:
-   * - when the cluster has processing documents in ESQueue before upgrading to v7.13 when ESQueue was removed
-   * - if Kibana crashes while a report task is executing and it couldn't be rescheduled on its own
-   *
-   * Pending reports are not included in this search: they may be scheduled in TM just not run yet.
-   * TODO Should we get a list of the reports that are pending and scheduled in TM so we can exclude them from this query?
+   * A report needs to be rescheduled when:
+   *   1. An older version of Kibana created jobs with ESQueue, and they have
+   *   not yet started running.
+   *   2. The report process_expiration field is overdue, which happens if the
+   *   report runs too long or Kibana restarts during execution
    */
-  public async findZombieReportDocuments(): Promise<ReportRecordTimeout[] | null> {
+  public async findStaleReportJob(): Promise<ReportRecordTimeout> {
     const client = await this.getClient();
+
+    const expiredFilter = {
+      bool: {
+        must: [
+          { range: { process_expiration: { lt: `now` } } },
+          { terms: { status: [statuses.JOB_STATUS_PROCESSING] } },
+        ],
+      },
+    };
+    const oldVersionFilter = {
+      bool: {
+        must: [{ terms: { status: [statuses.JOB_STATUS_PENDING] } }],
+        must_not: [{ exists: { field: 'migration_version' } }],
+      },
+    };
+
     const { body } = await client.search<ReportRecordTimeout['_source']>({
+      size: 1,
       index: this.indexPrefix + '-*',
-      filter_path: 'hits.hits',
+      seq_no_primary_term: true,
+      _source_excludes: ['output'],
       body: {
-        sort: { created_at: { order: 'desc' } },
-        query: {
-          bool: {
-            filter: [
-              {
-                bool: {
-                  must: [
-                    { range: { process_expiration: { lt: `now-${this.queueTimeoutMins}m` } } },
```

Comment:
This fixes an issue where the search for timed out jobs would wait until they were overdue + the queue timeout time. Now, it finds them when they are overdue, with no extra wait time added.